### PR TITLE
allowing for max outbound connection configuration

### PR DESCRIPTION
**Pull request by jramos**
This allows for the maximum number of connections to be configured in the settings UI.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -6,4 +6,6 @@ src/dist
 src/.project
 src/.pydevproject
 src/.settings/
-*.dll
\ No newline at end of file
+*.dll
+src/**/*.o
+src/**/*.so
```
Comment (PeterSurda) [inline .gitignore]: please change *.dll to src/**/*.dll

File: src/class_outgoingSynSender.py [hunk 2]
```diff
@@ -79,7 +79,7 @@ def run(self):
             self.stop.wait(2)
         while BMConfigParser().safeGetBoolean('bitmessagesettings', 'sendoutgoingconnections') and not self._stopped:
             self.name = "outgoingSynSender"
-            maximumConnections = 1 if state.trustedPeer else 8 # maximum number of outgoing connections = 8
+            maximumConnections = 1 if state.trustedPeer else BMConfigParser().getint('bitmessagesettings', 'maxoutboundconnections')
```
Comment (PeterSurda) [inline src/class_outgoingSynSender.py]: please change getint to safeGetInt

File: src/protocol.py [hunk 3]
```diff
@@ -495,3 +494,5 @@ def broadcastToSendDataQueues(data):
     # this means TLSv1, there is no way to set "TLSv1 or higher" or
     # "TLSv1.2" in < 2.7.9
     sslProtocolVersion = ssl.PROTOCOL_TLSv1
+
```
Comment (PeterSurda) [inline src/protocol.py]: please remove this whole hunk, this has been fixed

File: src/shared.py [hunk 4]
```diff
@@ -30,7 +30,6 @@
 from helper_sql import *
 from helper_threading import *
 from inventory import Inventory
-import protocol
 import state
```
Comment (PeterSurda) [inline src/shared.py]: please remove this while hunk as well, this has been fixed as well

**Comment by PeterSurda** [comment]
If you do the minor changes I commented upon inside the code, it's good to go, I just need to test if it works, which should be very quick. Please also rebase and PGP sign your commit, and when it's accepted, you can register here: https://tip4commit.com/github/Bitmessage/PyBitmessage and collect a tip. But it must be rebased and PGP signed so that I can do a fast-forward merge. Don't forget to add your PGP pubkey into your github profile.

**Comment by jramos** [comment]
Sure thing. I'll get these changes made soon.

**Comment by jramos** [comment]
@PeterSurda done.

**Comment by PeterSurda** [comment]
Thanks. PGP signature looks ok. Landscape.io however complains that ValueException is not valid, my bad, it should be ValueError.

**Pull request merged**